src/plugins/links/public/embeddable/links_embeddable.tsx

```diff
@@ -6,155 +6,271 @@
+              }
+            } catch {
+              // do nothing, user cancelled
+            }
+          },
+        },
+        {
+          ...titleComparators,
+          links: [
+            links$,
+            (nextLinks?: ResolvedLink[]) => links$.next(nextLinks ?? []),
+            (a, b) => Boolean(savedObjectId$.value) || fastIsEqual(a, b), // Editing attributes in a by-reference panel should not trigger unsaved changes.
+          ],
+          layout: [
+            layout$,
+            (nextLayout?: LinksLayoutType) => layout$.next(nextLayout ?? LINKS_VERTICAL_LAYOUT),
+            (a, b) => Boolean(savedObjectId$.value) || a === b,
+          ],
+          error: [error$, (nextError?: Error) => error$.next(nextError)],
+          defaultPanelDescription: [
+            defaultPanelDescription,
+            (nextDescription?: string) => defaultPanelDescription.next(nextDescription),
+          ],
+          defaultPanelTitle: [
+            defaultPanelTitle,
+            (nextTitle?: string) => defaultPanelTitle.next(nextTitle),
+          ],
+          savedObjectId: [savedObjectId$, (val) => savedObjectId$.next(val)],
+        }
+      );
+
+      const Component = () => {
+        const [links, layout] = useBatchedOptionalPublishingSubjects(links$, layout$);
+
+        const linkItems: { [id: string]: { id: string; content: JSX.Element } } = useMemo(() => {
+          if (!links) return {};
+          return links.reduce((prev, currentLink) => {
+            return {
+              ...prev,
+              [currentLink.id]: {
+                id: currentLink.id,
+                content:
+                  currentLink.type === DASHBOARD_LINK_TYPE ? (
+                    <DashboardLinkComponent
+                      key={currentLink.id}
+                      link={currentLink}
+                      layout={layout ?? LINKS_VERTICAL_LAYOUT}
+                      parentApi={parentApi as LinksParentApi}
+                    />
+                  ) : (
+                    <ExternalLinkComponent
+                      key={currentLink.id}
+                      link={currentLink}
+                      layout={layout ?? LINKS_VERTICAL_LAYOUT}
+                    />
+                  ),
+              },
+            };
+          }, {});
+        }, [links, layout]);
+        return (
+          <EuiPanel
+            className={`linksComponent ${
+              layout === LINKS_HORIZONTAL_LAYOUT ? 'eui-xScroll' : 'eui-yScroll'
+            }`}
+            paddingSize="xs"
+            data-shared-item
+            data-rendering-count={1}
+            data-test-subj="links--component"
+          >
+            <EuiListGroup
+              maxWidth={false}
+              className={`${layout ?? LINKS_VERTICAL_LAYOUT}LayoutWrapper`}
+              data-test-subj="links--component--listGroup"
+            >
+              {links?.map((link) => linkItems[link.id].content)}
+            </EuiListGroup>
```

Comment:
Not sure why, but something funky is happening with the styling and there is a bunch of extra spacing around the links:

![May-08-2024 12-45-48](https://github.com/elastic/kibana/assets/8698078/a291869f-4bf3-4d1d-bb48-79370f3c8f15)
    
![May-08-2024 12-48-26](https://github.com/elastic/kibana/assets/8698078/c8b84fd9-8900-46eb-9946-2b1b420248d5)
    
(green background is "before", blue is "after")

Reply:
Fixed in bf4b1c23bc9c6006a0c1b812bdc3a270fa94a6b1. Thanks for catching this.